src/guest_memory.rs

```diff
@@ -246,6 +246,22 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     unsafe fn as_mut_slice(&self) -> Option<&mut [u8]> {
         None
     }
+
+    /// Returns a [`VolatileSlice`](struct.VolatileSlice.html) of `count` bytes starting at
+    /// `offset`.
+    fn get_slice(
+        &self,
+        _offset: MemoryRegionAddress,
+        _count: usize,
+    ) -> Result<volatile_memory::VolatileSlice> {
+        Err(Error::HostAddressNotAvailable)
+    }
+
+    /// Gets a slice of memory for the entire region that supports volatile access.
+    fn as_volatile_slice(&self) -> volatile_memory::VolatileSlice {
+        self.get_slice(MemoryRegionAddress(0), self.len() as usize)
```

Comment:
Not sure you can have this implementation in the trait because this unwrap being safe depends on the implementation of `get_slice`. Is there any other contract which makes this unwrap safe?

Reply:
You're right, the unwrap needs to be removed.

Reply:
To remove the unwrap, do you mean changing the return type to `Result<volatile_memory::VolatileSlice>` ?

Reply:
Yes.